src/main/scala/com/cloudera/sparkts/models/KNNRegression.scala

```diff
@@ -0,0 +1,170 @@
+/**
+  * Copyright (C) 2017 Verizon. All Rights Reserved.
+  *
+  * Licensed under the Apache License, Version 2.0 (the "License");
+  * you may not use this file except in compliance with the License.
+  * You may obtain a copy of the License at
+  *     http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
+package com.cloudera.sparkts.models
+
+import org.apache.spark.mllib.KernelType._
+import org.apache.spark.mllib.Kernel
+import org.apache.spark.mllib.linalg.{Vector, Vectors}
+import scala.collection.mutable.ArrayBuffer
+import java.util.PriorityQueue
+import java.util.Comparator
+import scala.collection.JavaConverters._
+
+class BoundedPriorityQueue[E](k: Int,
```

Comment:
Spark has a BoundedPriorityQueue implementation - would it make sense to rely on that? https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/util/BoundedPriorityQueue.scala

Reply:
Yeah for BLAS I had to add the spark package and I can use the BoundedPriorityQueue from there...